tree/ntuple/v7/src/RNTupleDescriptor.cxx

```diff
@@ -707,6 +712,35 @@ std::unique_ptr<ROOT::Experimental::RNTupleModel> ROOT::Experimental::RNTupleDes
    return model;
 }
 
+ROOT::Experimental::RNTupleDescriptor::ENTupleMergeable
+ROOT::Experimental::RNTupleDescriptor::IsMergeable(const RNTupleDescriptor &other) const {
```

Comment:
This is just a quick usage sketch for discussion, we can omit it from the final PR.

Reply:
I think I'd find it more intuitive to have a recursive method that takes two field descriptors and check if they are compatible.  The initial invocation happens to be with the two top-level fields.

Reply:
Actually, if `IsMergable` works recursively on fields, we would not need a tree iterator for the field descriptor, but the simple iteration of the direct children is sufficient.

Reply:
In addition, iteration over child fields elsewhere can use this implementation, e.g. 
https://github.com/root-project/root/blob/aad714fec4012b2558d1e039243361d8573f998b/tree/ntuple/v7/src/RNTupleDescriptor.cxx#L701-L706

would become: 
```cpp 
for (const auto& topDesc : GetTopLevelFields()) { 
   auto field = Detail::RFieldBase::Create(topDesc.GetFieldName(), topDesc.GetTypeName());
   model->AddField(std::unique_ptr<Detail::RFieldBase>(field)); 
}                       
``` 
modulo the name changes going on for top-level / first-level fields.